hvsock.go

```diff
@@ -257,13 +507,19 @@ func (conn *HvsockConn) IsClosed() bool {
 	return conn.sock.IsClosed()
 }
 
+// shutdown disables sending or receiving on a socket
 func (conn *HvsockConn) shutdown(how int) error {
 	if conn.IsClosed() {
-		return ErrFileClosed
+		return sockets.ErrSocketClosed
 	}
 
 	err := syscall.Shutdown(conn.sock.handle, how)
 	if err != nil {
+		// If the connection was closed, shutdowns fail with "not connected"
+		if errors.Is(err, windows.WSAENOTCONN) ||
+			errors.Is(err, windows.WSAESHUTDOWN) {
+			err = sockets.ErrSocketClosed
```

Comment:
I don't like discarding the actual error msg from the OS, in some cases it can be extremely helpful in tracking down what actually happened. I'd see if we can come up with a custom error of our own that preserves this data.

Reply:
We could define our own error type/instance for socket closed that wraps it, but I was thinking that most users would not care about the actual error code number, just that the socket was closed. And it makes it easier to check with `errors.Is(err, net.ErrClosed)` or `errors.Is(err, sockets.ErrSocketClosed)`

I have been thinking about moving the error code listings from hcsshim here, so we may add a pkg\errdefs` here that lets us accomplish both...